Change frontend for vite+vue3 and tailwind using i18n for languages

mocenslabs opened the pull request:
Summary:
This PR introduces several improvements to the Profile-Pilot project’s frontend UI and functionality, focused on enhancing user experience and internationalization support.

Changes included:

    Refactored the Report.vue component to render GitHub profile analysis markdown properly using markdown-it, improving readability and visual style with Tailwind Typography.

    Added robust section parsing to split the report into clear, styled sections with translated section titles when the app language is Spanish.

    Improved error handling and loading states in App.vue during API calls.

    Updated Actions.vue component to copy and download the raw markdown report correctly without affecting UI rendering.

    Integrated i18n support for static UI texts, including a new localized footer component (Footer.vue).

    Adjusted layout spacing to improve visual separation between header, dark mode/language toggle, main content, and footer.

    Ensured TailwindCSS typography plugin is correctly configured and imported for consistent styling.

Benefits:

    Users get a cleaner, more readable report presentation.

    Spanish-speaking users see section titles translated.

    Overall UI is more polished, accessible, and responsive.

    Codebase is modular and easier to maintain and extend.

mocenslabs committed:
Change frontend for vite+vue3 and tailwind using i18n for languages

Ziqian-Huang0607 commented:
<img width="313" height="849" alt="Screenshot 2025-08-03 at 13 37 41" src="https://github.com/user-attachments/assets/87bf537f-b76c-449b-963d-0e849b5a9135" />
Hi, Really glad to see your PR, but there are somethings you might need to change before merge, as you see in the image, your language setting didn't change the language setting of the summarization but only the language of the buttons. To change language language of summarization you might have to edit index.py backend. Once you have done that and I have tested it works successful I will merge.

mocenslabs committed:
Add language support to API call and prepare i18n prompts

mocenslabs commented:
Summary:

This PR adds dynamic language support to the GitHub profile analyzer app:

    Passes selected language (en or es) as a query parameter in the API call.

    Implements locale-based prompt files (en.txt, es.txt) on the backend.

    Refactors the frontend App.vue to use vue-i18n to determine active language.

    UI improvements: better layout spacing for language/dark mode toggles, cleaner markdown rendering, and robust error handling.

Note:
As the backend cannot be tested locally, the lang parameter should now be received in the API and used to load localized prompts. Please test in production or staging.

mocenslabs committed:
feat: improve footer with dynamic year and i18n credits + load API key from .env in index.py

mocenslabs commented:
### Changes made:

#### 🌐 Frontend (Vue):
- Enhanced `<Footer />` component:
  - Added dynamic year using `new Date().getFullYear()`
  - Localized credit text via `i18n` (`en.json` and `es.json`)
  - Included links to:
    - Original project creator (Ziqian Huang)
    - My personal portfolio

#### 🧠 Backend (index.py):
- Integrated `.env` support using `os.getenv`
- Now uses `GOOGLE_API_KEY` from environment variables instead of hardcoding it
- Added a mock fallback message if the API key is missing

Let me know if any adjustments are needed for wording, style, or logic.

mocenslabs committed:
change url in fetch on App.vue

mocenslabs commented:
Send you the change of url i forgot change it for the original.

mocenslabs committed:
change url for api
